### PR TITLE
[Modules] Candidate List to display candidate consent

### PR DESCRIPTION
**Pull request by llevitis**
Rebasing PR #2755 proved to be messy, so I figured opening a new PR would be easier. The functionality is the same as that of the previous PR: bg-error is used to highlight a row in red if 'Consent to Study' in the Config module of config.xml for that participant is 'no'.

### CODE REVIEW

File: modules/candidate_list/js/columnFormatter.js [hunk 1]
```diff
@@ -1,2 +1,44 @@
...
+
+    function formatColumn(column, cell, rowData, rowHeaders) {
+        if (loris.hiddenHeaders.indexOf(column) > -1) return null;
+        var classes = [];
+        var row = {};
+        rowHeaders.forEach(function(header, index) {
+            row[header] = rowData[index]
+        }, this);
+        if ("PSCID" === column) {
+            var url = loris.BaseURL + "/" + rowData[1] + "/";
```
Comment (xlecours) [inline modules/candidate_list/js/columnFormatter.js]: This should use `row['DCCID']` now that the row variable is available (line # 27 in the jsx file)

File: modules/candidate_list/php/candidate_list.class.inc [hunk 2]
```diff
@@ -78,6 +79,35 @@ class Candidate_List extends \NDB_Menu_Filter
...
+            //   array elements is returned
+            //We want just the numeric array of assoc array elements all the time.
+
+            if (!isset($consent_details[0])) {
+                $consent_details = \Utility::toArray($consent_details);
+            }
+
+            $consentType = $consent_details[0];
+            $name        = $db->escape($consentType["name"]);
+            $label       = $db->escape($consentType["label"]);
```
Comment (xlecours) [inline modules/candidate_list/php/candidate_list.class.inc]: `$label` is not used and should be removed

**Comment by ridz1208** [comment]
@llevitis please attach a before/after image to show how it will look.

Also confirm with @LeighMac that this is indeed what she wants to see

**Comment by llevitis** [comment]
@ridz1208 Here's a screenshot: the 2 candidates highlighted in red have 'Consent To Study' set to 'no'.  

image

**Comment by ridz1208** [comment]
@llevitis so what happens if you have multiple consents??
and why is there not coloring on 2 of the columns ?

**Comment by llevitis** [comment]
@ridz1208 I emailed Leigh to confirm that this is what she wants to see, but over the summer she said that "Study to Consent" is the most important one. I'll check why the highlighting isn't happening for columns 'DCCID' & 'Scan Done' - presumably the fact that they're links takes precedence.

**Comment by ridz1208** [comment]
@llevitis yeah but this is very case specific cause the "consent to study" is just a name it could very well be another name since its in the config.xml

This will be project specific

**Comment by ridz1208** [comment]
This is partly approved and waiting on https://github.com/aces/Loris/pull/3263 for normalised tables

**Comment by llevitis** [comment]
'Feedback' is the only column to which bg-error can't be applied now since it already has background colors specified for the different types of feedback, e.g.: 

image

**Comment by driusan** [comment]
@llevitis I think this should be closed until the consent structure is properly normalized and there are design decisions made on things like how to properly deal with LORIS setups that have multiple consent types defined. Otherwise this is going to just keep having conflicts, bikeshedding, and rebases.

**Comment by llevitis** [comment]
@driusan - sounds good, I'll close it.

**Pull request closed without merging**